### ISSUE TITLE
Improve introduction event match block

### ISSUE THREAD

**Issue by regexident** [rustbridge/Rusty-Snake-Book#35]
There's a lot to digest in this snippet …

```rust
match event {
    Event::KeyDown {
        keycode: Some(Keycode::Escape),
        ..
    } => break 'game,
    _ => continue 'game,
}
```

With barely any prior introduction to the features shown it ends up hitting newcomers to Rust in the face like a wrecking ball.

It got them all:

- `enum` with associated values.
- multi-level destructuring
- `Some(…)` <sup>(which hides an `Option::Some` by eliding the `Option::`)</sup>
- `..`
- `'game` <sup>(I can count the number of times I've seen a label like this be used in the wild on one hand)</sup>
- `break` vs. `continue`
- `_`
- `=>`

To make things worse: with the exception of "`break` vs. `continue`" none of these things can easily be searched for either.

The `continue 'game` should probably also be just `continue`, as else one could end up dropping events, that should be handled.

--

Over all it might make sense to add links to corresponding references in [Rust by Example](https://doc.rust-lang.org/rust-by-example) when introducing new concepts and syntax.

**Comment by jonas-schievink** [comment]
Yeah, that's a lot to digest all at once, though my ggez rewrite obsoletes that code
